shylock/backends/pymemcache.py

```diff
@@ -0,0 +1,95 @@
+from time import sleep, time
+from typing import Optional
+
+try:
+    from pymemcache import Client, MemcacheError
+except ImportError:
+    Client = None
+    MemcacheError = None
+
+from shylock.backends import ShylockSyncBackend
+from shylock.exceptions import ShylockException
+
+DOCUMENT_TTL = 60 * 5  # 5min seems like a reasonable TTL
+POLL_DELAY = 1 / 16  # Some balance between high polling and high delay
+RETRY_COUNT = 3  # If a memcache error occurs, retry for this many times.
+
+
+class ShylockPymemcacheBackend(ShylockSyncBackend):
+    def __init__(
+        self,
+        client: Client,
+        noreply: Optional[bool] = False,
+        flags: Optional[int] = None,
+    ):
+        self._check()
+        self._client = client
+        self._noreply = noreply
+        self._flags = flags
+        self._owner = None
+
+    @staticmethod
+    def create(
+        client: Client, noreply: Optional[bool] = False, flags: Optional[int] = None
+    ) -> "ShylockPymemcacheBackend":
+        """
+        Create and initialize the backend
+        :param client: An instance of pymemcache.Client connected to the desired server.
+        :param noreply: If True, do not wait for a reply.
```

Comment:
Quickly looking at the docs it seems that `noreply` means `acquire` is unable to confirm if it was able to get the lock or not. You're sure this works on multiple processes?